### PR TITLE
Merge 4.6 release branch to master

### PR DESCRIPTION
**Pull request by karuturi**
Initial merge of 4.6 to master
ignored pom.xml version number changes and changes to debian/changelog and engine/schema/src/com/cloud/upgrade/DatabaseUpgradeChecker.java

Following commands were executed
```
1. git checkout 4.6
2. git pull --rebase
3. git checkout master
4. git pull --rebase
5. git fwd-merge 4.6
6. git diff --name-only | grep pom.xml | xargs git checkout --ours
7. git diff --name-only | grep pom.xml | xargs git add
8. git checkout --ours engine/schema/src/com/cloud/upgrade/DatabaseUpgradeChecker.java
9. git add engine/schema/src/com/cloud/upgrade/DatabaseUpgradeChecker.java
10. git checkout --ours debian/changelog
11. git add debian/changelog
12. # manually edited version number in tools/marvin/marvin/deployAndRun.py and tools/marvin/setup.py
13. git commit
14. git checkout -b "merge-46-to-master"
```

### CODE REVIEW

File: build/replace.properties [hunk 1]
```diff
@@ -26,4 +26,4 @@ AGENTLOG=logs/agent.log
 MSMNTDIR=/mnt
 COMPONENTS-SPEC=components.xml
 REMOTEHOST=localhost
-COMMONLIBDIR=client/target/cloud-client-ui-4.6.0-SNAPSHOT/WEB-INF/lib/
+COMMONLIBDIR=client/target/cloud-client-ui-4.6.0/WEB-INF/lib/
```
Comment (remibergsma) [inline build/replace.properties]: @karuturi Shouldn't this be 4.7.0-SNAPSHOT?

**Comment by remibergsma** [comment]
Thanks @karuturi will give it a test-drive soon!

File: build/replace.properties (hunk 1 above)
Comment (karuturi) [inline build/replace.properties]: yes. will update.

**Comment by remibergsma** [comment]
@DaanHoogland @wilderrodrigues @miguelaferreira Can either of you review so we can merge this and open master for new features?

**Comment by wilderrodrigues** [comment]
Hi @remibergsma and @karuturi 

Went through the code, which was quite straight forward. Based on that, this PR LGTM :+1: 

Cheers,
Wilder

**Comment by DaanHoogland** [comment]
Had  a look yesterday and lgtm, didn't want to comment untill I saw the test results. (It seemed so small ;)

**Pull request merged**